src/wp-includes/functions.php

```diff
@@ -6998,6 +6998,27 @@ function wp_debug_backtrace_summary( $ignore_class = null, $skip_frames = 0, $pr
  * @return int[] Array of IDs not present in the cache.
  */
 function _get_non_cached_ids( $object_ids, $cache_key ) {
+	$object_ids_filter = static function ( $object_id ) {
+		/*
+		 * This treats both `16` and `"16"` as valid for the purposes of the cache key. Other
+		 * numeric types and numeric strings are discarded, for example `16.3` and `"16.3"`.
+		 *
+		 * This could all be achieved with filter_var(), however the filter extension can be removed
+		 * in custom builds so WordPress is unable to rely on it.
+		 */
+		// phpcs:ignore WordPress.PHP.YodaConditions.NotYoda, false positive
+		if ( is_int( $object_id ) || ( is_string( $object_id ) && (string) $object_id === (string) (int) $object_id ) ) {
```

Comment:
```suggestion
		if ( is_int( $object_id )
			|| ( is_string( $object_id ) && (string) (int) $object_id === $object_id 
```
We don't need the `phpcs:ignore` comment if we flip the order around, I get no errors from WPCS then.

We can also remove redundant type casting on one side, as we already have the `is_string()` check directly before.